Fix ban system

SignatureBeef opened the pull request:
This improves the migration/conversion process in the ban manager for both sqlite & mysql; and on the side it allows TShock to build on OSX/MonoDevelop (assembly name fix, not a big deal).

My end i have confirmed that the server now boots for both database implementations (on OSX).
I have not yet tested to make sure the ban system actually works in game etc.

SignatureBeef committed:
Fix OTAPI reference

This now allows MonoDevelop to compile without receiving error while starting a build: System.IO.FileLoadException: The assembly name is invalid

SignatureBeef committed:
Fix mysql & sqlite ban manager migration during init

Tested with existing bans in both sqlite & mysql environments.
This is in addition to the syntax fix from earlier.

SignatureBeef committed:
Update CHANGELOG.md

QuiCM commented:
This looks fine to me

hakusaro commented:
oml

hakusaro commented:
I merged #2269 first because I didn't realize that the base ref of this PR was #2269's branch. Oops. I can't test the ban conversion process because I don't have a MySQL server configured and I don't have an old ban database in either format, so I'm waiting for someone to test the conversion process.

SignatureBeef committed:
Corrected UUID InsertBan in TSPlayer

Pointed out thanks to this discussion: https://github.com/Pryaxis/TShock/discussions/2268#discussioncomment-598331

QuiCM commented:
Can confirm the conversion works on MySQL 8.0.23

QuiCM commented:
Also seems to work with SQLite
